test/common.py

```diff
@@ -2357,6 +2357,8 @@ def run_browser(self, html_file, expected=None, message=None, timeout=None, extr
             if extra_tries > 0:
               print('[test error (see below), automatically retrying]')
               print(e)
+              if '?' not in html_file:
+                html_file += '?capture_stdio'
```

Comment:
Worth a comment or a debug logging here ("enabling stdio capture for debugging")